scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java

```diff
@@ -66,7 +68,35 @@
 
 	private final AtomicReference<Handshaker> ongoingHandshake = new AtomicReference<Handshaker>();
 	private final SessionListener sessionListener = new ConnectionSessionListener();
+	private final ExecutionListener executionListener = new ExecutionListener() {
 
+		@Override
+		public void beforeExecution() {
+			if (peerAddress != null) {
+				MDC.put("PEER", StringUtil.toString(peerAddress));
+			}
+			if (cid != null) {
+				MDC.put("CONNECTION_ID", cid.getAsString());
+			}
+			SessionId sid = sessionId;
+			DTLSSession session = getSession();
+			if (session != null) {
+				sid = session.getSessionIdentifier();
+				ConnectionId writeConnectionId = session.getWriteConnectionId();
+				if (writeConnectionId != null) {
+					MDC.put("WRITE_CONNECTION_ID", writeConnectionId.getAsString());
```

Comment:
In the log appears this `, WRITE_CONNECTION_ID=} ` a check String.isEmpty() would fix it.

Reply:
OK. I adapt it.